src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -108,17 +100,18 @@ private CollectionNode GetCollectionOperandFromToken(QueryToken queryToken, IEdm
                     if (expectedTypeFullName.Equals("Edm.String"))

```

Comment:
>if (expectedTypeFullName.Equals("Edm.String")) [](start = 20, length = 46)

Please open an issue to track the fact that this logic doesn't support use of in with a Collection(Edm.Primitive). Admittedly an edge case, and not what this issue is trying to fix, but something that we should track in case we need to support it in the future.

Reply:
Done at: [https://github.com/OData/odata.net/issues/1644](https://github.com/OData/odata.net/issues/1644)

---
In reply to: [370718208](https://github.com/OData/odata.net/pull/1642#discussion_r370718208) [](ancestors = 370718208)